### PR TITLE
feat: add .c8ignore support for filtering deploy/watch file scanning

### LINKED ISSUE #161: introduce .c8ignore for ignoring certain directories for artifact interops
the "usual suspects": `target/**/*`, `node_modules/**/*`, `.git/**/*` so that fex `c8 deploy` or `c8 watch` doesn't pick up Zeebe artifacts from these dirs

### PR DESCRIPTION
**Pull request by jwulf**
## Summary

Introduces `.c8ignore` file support so that `deploy` (directory scan) and `watch` (file monitoring) can filter out unwanted directories and files. Uses `.gitignore`-style pattern syntax via the [`ignore`](https://www.npmjs.com/package/ignore) npm package.

Closes #161

## Default ignore patterns (always active)

These directories are ignored automatically, with no `.c8ignore` file needed:

- `node_modules/`
- `target/`
- `.git/`

## Custom patterns via `.c8ignore`

Users create a `.c8ignore` file in their project root:

```gitignore
# Ignore build output
dist/
build/

# Ignore draft processes
**/draft-*.bpmn

# But keep this specific one
!draft-approved.bpmn
```

The file supports the full `.gitignore` syntax: glob patterns, directory trailing slashes, comments (`#`), blank lines, and negation (`!`) for file-level patterns.

Custom patterns are merged with the built-in defaults — they do not replace them.

## Changes

- **`src/ignore.ts`** (new) — `loadIgnoreRules(baseDir)` loads default patterns + `.c8ignore`; `isIgnored(ig, path, baseDir)` checks whether a path matches
- **`src/commands/deployments.ts`** — `collectResourceFiles()` now accepts an `Ignore` instance; `deploy()` loads ignore rules from `process.cwd()` and passes them through traversal. Ignored directories are skipped early to avoid descending into them.
- **`src/commands/watch.ts`** — File change events are filtered through ignore rules before triggering deploys
- **`package.json`** — Added `ignore` runtime dependency
- **`README.md`** — Feature bullet + `.c8ignore` section with example
- **`EXAMPLES.md`** — `.c8ignore` section with pattern examples

## Tests

22 new unit tests in `tests/unit/c8ignore.test.ts` covering:

| Category | Tests |
|---|---|
| Default ignore patterns | `node_modules/`, `target/`, `.git/` ignored without `.c8ignore` file |
| `.c8ignore` patterns | Glob directives, wildcard patterns, negation, comments, blank lines, additive with defaults, nested paths |
| Edge cases | No `.c8ignore` file, empty `.c8ignore`, building block folders, explicit BB ignore, all resource extensions (`.bpmn`/`.dmn`/`.form`), specific file ignore |
| Unit tests | `loadIgnoreRules()`, `isIgnored()` direct function testing |

All 497 unit tests pass. Build compiles cleanly.

### CODE REVIEW

File: src/commands/watch.ts [hunk 1]
```diff
@@ -63,6 +67,11 @@ export async function watchFiles(paths: string[], options: {
 
       const fullPath = isDirectory ? resolve(path, filename) : path;
 
+      // Skip files matching .c8ignore rules
+      if (isIgnored(ig, fullPath, resolve(process.cwd()))) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (jwulf) [reply src/commands/watch.ts:74]: Added a watch-mode test in `tests/unit/c8ignore.test.ts` under "watch mode respects .c8ignore". It creates a `.c8ignore` with `ignored/`, spawns watch (cwd = testDir), writes to `ignored/hidden.bpmn`, waits through debounce+cooldown, and asserts "Change detected" does NOT appear. 498 tests pass total.

File: src/ignore.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * Returns `true` when the path matches an ignore rule.
+ */
+export function isIgnored(ig: Ignore, filePath: string, baseDir: string): boolean {
+  let rel = relative(baseDir, filePath);
+  // `ignore` expects forward-slash separators
+  if (sep !== '/') {
+    rel = rel.split(sep).join('/');
+  }
+  // Paths outside baseDir can't be ignored
+  if (rel.startsWith('..') || rel === '') {
```
[the remark this answers is not in the document]
Comment (jwulf) [reply src/ignore.ts]: Fixed — adopted the suggested separator-aware check: `if (rel === '' || rel === '..' || rel.startsWith('../'))`. This correctly handles edge cases like `..foo.bpmn` filenames.

File: tests/unit/c8ignore.test.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+ * the resources that would be deployed.
+ */
+function dryRunDeploy(cwd: string, paths: string[] = ['.']): { body: { resources: { name: string }[] } } {
+  const result = spawnSync('node', [
+    '--experimental-strip-types',
+    CLI_ENTRY,
+    'deploy', ...paths,
+    '--dry-run',
+    '--output', 'json',
+  ], {
```
[the remark this answers is not in the document]
Comment (jwulf) [reply tests/unit/c8ignore.test.ts:25]: Fixed — removed `--output json` from the test helper. Dry-run always emits JSON regardless, and `--output` was being consumed as an unknown boolean flag (with `strict: false`) leaving `json` as a stray positional that could pollute deploy paths.

File: src/commands/deployments.ts [hunk 4]
```diff
@@ -137,6 +142,11 @@ function collectResourceFiles(dirPath: string, collected: ResourceFile[] = [], b
   }
 
   if (stat.isDirectory()) {
+    // Skip entire directory if it matches an ignore rule
+    if (ig && isIgnored(ig, dirPath + '/', basePath)) {
+      return collected;
```
[the remark this answers is not in the document]
Comment (jwulf) [reply src/commands/deployments.ts:147]: Fixed — `collectResourceFiles` now takes a separate `ignoreBaseDir` parameter (the cwd where `.c8ignore` was loaded from), decoupled from `basePath` (the scan root used for building-block/PA detection). All `isIgnored()` calls use `ignoreBaseDir`, ensuring patterns are always evaluated relative to cwd regardless of which directory is being scanned.

File: src/commands/watch.ts [hunk 5]
```diff
@@ -63,6 +67,11 @@ export async function watchFiles(paths: string[], options: {
 
       const fullPath = isDirectory ? resolve(path, filename) : path;
 
+      // Skip files matching .c8ignore rules
+      if (isIgnored(ig, fullPath, resolve(process.cwd()))) {
+        return;
```
[the remark this answers is not in the document]
Comment (jwulf) [reply src/commands/watch.ts:73]: Fixed — hoisted to `const ignoreBaseDir = resolve(process.cwd())` at the top; both `loadIgnoreRules(ignoreBaseDir)` and `isIgnored(ig, fullPath, ignoreBaseDir)` use the same constant.
